### PR TITLE
deprecate old index types and non-object stores

### PR DESCRIPTION
**Pull request by ashwanthgoli**
**What this PR does / why we need it**:
Deprecates old index types: `aws, aws-dynamo, gcp, gcp-columnkey, bigtable, bigtable-hashed, cassandra, grpc`. Only `tsdb` and `boltdb-shipper` are supported.

Deprecates non-object stores. The next major release might not allow the usage of these as `object_store` in `schema_config`: `aws-dynamo, gcp, gcp-columnkey, bigtable, bigtable-hashed, cassandra, grpc.`

Deprecates table manager which is not needed once when using `boltdb-shipper` or `tsdb` index type

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

**Comment by chaudum** [comment]
What about also deprecating `boltdb` (not `boltdb-shipper`)?

**Review by kavirajk — APPROVED** [review]
🚀 LGTM. thanks @ashwanthgoli 

cc @JStickler

**Comment by ashwanthgoli** [comment]
> What about also deprecating `boltdb` (not `boltdb-shipper`)?

@chaudum added it with https://github.com/grafana/loki/pull/10341/commits/a042be31d6d9348a5531d617381afe3d05a8f9b0

File: pkg/loki/modules.go [hunk 1]
```diff
@@ -507,6 +507,8 @@ func (t *Loki) initIngester() (_ services.Service, err error) {
 }
 
 func (t *Loki) initTableManager() (services.Service, error) {
+	level.Warn(util_log.Logger).Log("msg", "table manager is deprecated. Consider migrating to tsdb index which relies on a compactor instead.")
```
Comment (JStickler) [inline pkg/loki/modules.go:510]: Nice addition.

**Review by JStickler — APPROVED** [review]
[Docs team]  Really appreciate the addition of all the warning messages, that's helpful for usability.

**Pull request merged**